### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by jucora**
Thanks for code review

### CODE REVIEW

File: spec/my_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+require '../my_methods'
```
Comment (maelfosso) [inline spec/my_methods_spec.rb]: **`REQUIRED`** Please use `require_relative`

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @jucora @Ashah15 

# :clap:  Thank for that beautiful RSPEC code :smile: 

The test all passed. But before moving forward there some changes you have to make.

- [x] Please use `require_relative` instead of `require`

- [x] Write the test by comparing the result not with a constant value but with the result of the original method
**For example** instead of writing
```ruby
expect(test_array.my_all? do |x| x % 2 == 0 end).to eq(true)
```
Write
```
expect(test_array.my_all? do |x| x % 2 == 0 end).to eq(test_array.all? do |x| x % 2 == 0 end)
```
- [x] Some methods like `my_all?`, `my_any?`, `my_count`, `my_inject` need more than just one test because there is a lot of use case. For example, **my_count**, there is a use case without an argument and another one with the block. So Please complete.

Kindly, make the suggested change and re-submit a new code review request.
Thanks

**Comment by jucora** [comment]
Thanks for code review, it was helpful

Now methods receive arguments and more tests were added and also all the expects for every test are compared with the result of the original method.

Thanks

File: spec/my_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
+require_relative '../my_methods'
```
Comment (samgaco) [inline spec/my_methods_spec.rb:1]: :+1:

File: spec/my_methods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+RSpec.describe Enumerable do
+  let(:test_array) { [2, 2, 2, 4, 8] }
+
+  let(:test_array_1) { [1, 2, 3, 4] }
+
+  let(:test_array_2) { [2, 2, 2, 5, 2] }
+
+  describe '#my_each' do
+    it 'returns each of the elements in the array' do
+      expect(test_array_1.my_each { |element| element }).to eql(test_array_1.each { |element| element })
```
Comment (samgaco) [inline spec/my_methods_spec.rb:12]: Well done, like this you are testing against the methods you are trying to emulate :100: :white_check_mark:

**Review by samgaco — APPROVED** [review]
Hi @Ashah15 and @jucora :high_brightness: 

Good job team :100: the project is approved, you may merge now to the master branch and submit the project completion form.

------------------

And remember :arrow_down: 

* **[For your next code review requests]**  Add a short summary of changes in Pull Request description


----------------


* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**